From: Devaraj Rangasamy <dev@ti.com>
Date: Thu, 28 Apr 2011 22:20:30 +0000 (+0530)
Subject: DSS HDMI: Fix for regular suspend/resume break in DB
X-Git-Tag: android-2.6.35-2.3-omap4.13.1~3^2
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=ae7e7cbdde5792f8c103662c96e8e5928b287123

DSS HDMI: Fix for regular suspend/resume break in DB

When system is suspended with HDMI connected (GUI/video),
DSS/CORE retention is not happening.

Fixed the clock disable logic to relinquish DSS optional clock,
so as to aid suspend transitions.

Tested HPD,suspend/resume scenarios.

Change-Id: I799a39bf24d07ef3a626b87d3f6bd026f96e645a
Signed-off-by: Devaraj Rangasamy <dev@ti.com>
---

diff --git a/drivers/video/omap2/dss/hdmi.c b/drivers/video/omap2/dss/hdmi.c
index 36b49fb..8c0d97b 100644
--- a/drivers/video/omap2/dss/hdmi.c
+++ b/drivers/video/omap2/dss/hdmi.c
@@ -1775,7 +1775,7 @@ static int hdmi_set_power(struct omap_dss_device *dssdev,
 			     state_need == HDMI_SUSPENDED ? 'S' : 'D'), r);
 
 	if (cpu_is_omap44xx())
-		if ((power_need == HDMI_POWER_MIN) && hdmi_opt_clk_state) {
+		if ((power_need < HDMI_POWER_FULL) && hdmi_opt_clk_state) {
 			DSSINFO("Disable HDMI_PHY_48MHz clock, "
 				"hdmi_power = %d\n\n", power_need);
 			hdmi_opt_clock_disable();